### PR TITLE
Corrected the selected field names displayed in the schema of JDBC formatted response

### PR DESCRIPTION
**Pull request by chloe-zh**
*Issue #, if available:*

* [**Issue #290 Fieldnames with functions in SELECT clause are not proper in the schema of JDBC format**](https://github.com/opendistro-for-elasticsearch/sql/issues/290)

*Description of changes:*

* Fixed the issue that names not displayed properly in the schema of JDBC format.
* Added IT.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java [hunk 1]
```diff
@@ -94,6 +95,9 @@ private Field makeFieldImpl(SQLExpr expr, String alias, String tableAlias) throw
                 return makeFilterMethodField(mExpr, alias);
             }
 
+            if ((SQLFunctions.builtInFunctions.contains(methodName.toLowerCase())) && Strings.isNullOrEmpty(alias)) {
```
Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java]: I wonder is the first check necessary? `(SQLFunctions.builtInFunctions.contains(methodName.toLowerCase()))`. Any edge case?

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java [hunk 2]
```diff
@@ -137,4 +137,32 @@ public void ipTypeShouldPassJdbcFormatter() {
                 containsString("\"type\": \"ip\"")
         );
     }
+
+    @Test
```
Comment (davidcui-amzn) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java]: is this case-sensitive? If my query contains "SELECT SUBSTRING(lastname, 1, 2)" will it return "substring" or "SUBSTRING"

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java [hunk 3]
```diff
@@ -137,4 +137,32 @@ public void ipTypeShouldPassJdbcFormatter() {
...
+        );
+        assertThat(
+                executeQuery("SELECT log(balance) FROM " + TestsConstants.TEST_INDEX_ACCOUNT
+                        + " ORDER BY log(balance)", "jdbc"),
+                containsString("\"name\": \"log(balance)\"")
+        );
+    }
+
+    @Test
+    public void functionWithAliasShoultHaveAliasAsNameInSchema() {
```
Comment (davidcui-amzn) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java]: minor typo here- Should vs. Shoult

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java [hunk 4]
```diff
@@ -137,4 +137,32 @@ public void ipTypeShouldPassJdbcFormatter() {
                 containsString("\"type\": \"ip\"")
         );
     }
+
+    @Test
+    public void functionWithoutAliasShouldHaveEntireFunctionAsNameInSchema() {
+        assertThat(
```
Comment (galkk) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java]: split to separate tests

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java (hunk 2 above)
Comment (chloe-zh) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/JdbcTestIT.java]: Yep it is. it returns SUBSTRING

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java (hunk 1 above)
Comment (chloe-zh) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java]: Yep, it was necessary. Some tests were broken, for example:
`
SELECT address, CASE WHEN gender='0' THEN 'aaa' ELSE 'bbb' END AS a2345, count(age) FROM accounts GROUP BY terms('field'='address','execution_hint'='global_ordinals'), a2345
`
For the part of "terms", what we expected was `"terms(field=address,execution_hint=global_ordinals)"`, what we actually got was `"terms(com.alibaba.druid.sql.ast.expr.SQLBinaryOpExpr@fedcc4,com.alibaba.druid.sql.ast.expr.SQLBinaryOpExpr@f90c3f62)"`

Comment (dai-chen) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java]: I see. Thanks!
